graph-manager-docs/source/schema-registry.mdx

```diff
@@ -5,23 +5,110 @@ description: A central hub for your data graph
+
+    ```none:title=.env
+    APOLLO_KEY=YOUR_KEY_HERE
+    ```
+
+3. Set the `engine.reportSchema` option to `true` in your `ApolloServer` constructor, like so:
+
+    ```js
+    const server = new ApolloServer({
+      // ...other options...
+      engine {
+        reportSchema: true
+      }
+    });
+    ```
+
+Now every time your server finishes starting up, it registers its schema with Graph Manager.
+
+> **Note:** After starting up, Apollo Server waits a random amount of time between zero and ten seconds before automatically registering its schema. [You can customize this behavior.](#customizing-behavior-in-apollo-server)
+
+#### Registering automatically to a variant
+
+Your server can register its schema to a particular [variant of your graph](#managing-environments-with-variants). Each of your server's environments (development, staging, production, etc.) should register to a different variant.
+
+To do so, set the `APOLLO_GRAPH_VARIANT` environment variable in each server environment:
+
+```none:title=.env
+APOLLO_KEY=YOUR_KEY_HERE
+APOLLO_GRAPH_VARIANT=staging
+```
+
+If you don't specify a variant, Apollo Server reports its schema to the default variant (named `current`).
+
+#### Customizing behavior in Apollo Server
+
+In addition to `engine.reportSchema`, you can provide the following options to Apollo Server to modify schema reporting defaults:
+
+| Name  | Type | Description  |
+|---|---|---|
+| `engine.overrideReportedSchema` | String | If provided, Apollo Server sends this string value as the schema to register. By default, Apollo Server sends its interpreted `GraphQLSchema` object (the same value it returns from an introspection request) as an SDL string. This default behavior is appropriate for the majority of use cases. |
+| `engine.schemaReportingInitialDelayMaxMs` | Integer | On startup, Apollo Server waits a random number of milliseconds between `0` and this value before attempting to register its schema. This helps stagger registration requests when you deploy multiple instances of your server simultaneously. The default value is `10000` (10 seconds). |
+
+For example, the following value for `engine.overrideReportedSchema` provides a schema definition directly from Apollo Server's `typeDefs` object, instead of the interpreted `GraphQLSchema` created from that object:
+
+```js{5}
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  engine {
+    reportSchema: true
+    overrideReportedSchema: typeDefs.loc && typeDefs.loc.source.body
+  }
+});
+```
+
+### Other GraphQL servers
+
+Any GraphQL server can perform automatic schema registration by implementing the [Schema reporting protocol](./schema-reporting-protocol/).
+
+> Currently, we are not aware of GraphQL server libraries besides Apollo Server that implement the schema reporting protocol. If you add support to an open-source library, please let us know at **support@apollographql.com**!
+
+### Handling conflicting schemas
```

Comment:
I'm worried that users who are trying to answer the question of "what happens when I have multiple servers w/ different schemas?" will find this (but are rare) and those who are wondering "what happens during a deployment of a new schema?" might not. Perhaps we could name this somewhat more generically, like "Breaking ties" or "Handling roll-out of a new schema"?